### PR TITLE
Firebase challengeNotifier trigger function

### PR DESCRIPTION
**Pull request by vince0656**

### CODE REVIEW

File: src/Apps/CheezeWizards/Services/Firebase/FirebaseService.js [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+                }
+                const wizards = [];
+                snapshots.docs.forEach(doc => {
+                    wizards.push(doc.data());
+                });
+                return wizards;
+            });
+    }
+
+    // Todo: should return an empty array if network is not defined or not a valid string + associated test
```
Comment (jamesmorgan) [inline src/Apps/CheezeWizards/Services/Firebase/FirebaseService.js]: Are these todos still valid?

File: src/Apps/CheezeWizards/Services/Firebase/FirebaseService.js [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
```
Comment (jamesmorgan) [inline src/Apps/CheezeWizards/Services/Firebase/FirebaseService.js]: Are these todos still valid?

File: src/Apps/CheezeWizards/Services/Firebase/FirebaseService.js [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+
+        checkValidNetwork(network);
+
+        const networkRef = db.collection('wizards').doc('network').collection(network);
+        const challengeRef = networkRef
+            .doc(wizardId)
+            .collection('duel')
+            .doc(challengeId);
+
+        await db.runTransaction(t => {
```
Comment (jamesmorgan) [inline src/Apps/CheezeWizards/Services/Firebase/FirebaseService.js]: Nice work of transactionality boundaries 👍

File: src/Apps/CheezeWizards/Services/Firebase/FirebaseService.js [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+
+    async allOutstandingChallenges(network, owner) {
+        // TODO I need this in the stage
+    }
+
+    async registerFirebaseMessagingTokenForEthAddress() {
+
+    }
+
+    // async registerWizardForDueling(network, {owner, wizard}) {
```
Comment (jamesmorgan) [inline src/Apps/CheezeWizards/Services/Firebase/FirebaseService.js]: Ditch old code which is no longer needed

File: __tests__/Apps/CheezeWizards/FirebaseService-test.js [hunk 5]
```diff
@@ -96,3 +123,39 @@ test.only("Can take all of an owner's wizards offline", async () => {
     expect(allWizardsOffline).toBe(true);
 });
 
+test.only("Can issue a challenge successfully", async () => {
```
Comment (andygray) [inline __tests__/Apps/CheezeWizards/FirebaseService-test.js]: drop `only`

File: src/Apps/CheezeWizards/Services/Firebase/FirebaseService.js (hunk 4 above)
Comment (vince0656) [reply src/Apps/CheezeWizards/Services/Firebase/FirebaseService.js]: I'm not sure about the comments - will need to speak with Andy about them

File: src/Apps/CheezeWizards/Services/Firebase/FirebaseService.js (hunk 3 above)
Comment (vince0656) [reply src/Apps/CheezeWizards/Services/Firebase/FirebaseService.js]: Yeah it was great to figure this out

File: src/Apps/CheezeWizards/Services/Firebase/FirebaseService.js (hunk 1 above)
Comment (vince0656) [reply src/Apps/CheezeWizards/Services/Firebase/FirebaseService.js]: I'll address these now

File: src/Apps/CheezeWizards/Services/Firebase/FirebaseService.js (hunk 2 above)
Comment (vince0656) [reply src/Apps/CheezeWizards/Services/Firebase/FirebaseService.js]: I'll address these now

**Pull request merged**